### PR TITLE
dt-shell-color-scripts: init at unstable-2022-02-22

### PR DESCRIPTION
**Pull request by jakobneufeld**
###### Description of changes

This is a collection of shell color scripts by Derek Taylor. Sorry for republishing PR but I messed up the commit history by accident. I think I have amended all the changes  that @AndersonTorres requested

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jakobneufeld** [comment]
Sorry again for reposting the PR

File: pkgs/tools/misc/dt-shell-color-scripts/default.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{ stdenv
+, lib
+, fetchFromGitLab
+, runtimeShell
+, bash
+, makeWrapper
+}:
+  stdenv.mkDerivation {
```
Comment (AndersonTorres) [inline pkgs/tools/misc/dt-shell-color-scripts/default.nix:8]:
```suggestion

stdenv.mkDerivation {
```

Remove this indentation.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]
Also, learn the basic Git commands, especially `rebase`.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/tools/misc/dt-shell-color-scripts/default.nix [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      rev = "fcd013ea2e1ff80e01adbcea9d0eaf9c73db94c0";
+      sha256 = "sha256-bd3NBf99rCiADUKQb6fzCBDaKBmYaZHcO4qokm/39do=";
+    };
+
+    nativeBuildInputs = [ makeWrapper ];
+
+
+
+    installPhase = ''
+    runHook preInstall
```
Comment (AndersonTorres) [inline pkgs/tools/misc/dt-shell-color-scripts/default.nix:27]:
```suggestion
      runHook preInstall
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by jakobneufeld** [comment]
What do you think

File: pkgs/tools/misc/dt-shell-color-scripts/default.nix [hunk 3]
```diff
@@ -0,0 +1,44 @@
+{ stdenv
+, lib
+, fetchFromGitLab
+, runtimeShell
+, bash
```
Comment (AndersonTorres) [inline pkgs/tools/misc/dt-shell-color-scripts/default.nix:5]:
```suggestion
```

Unneeded input.

File: pkgs/tools/misc/dt-shell-color-scripts/default.nix [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    mkdir -p $out/opt/shell-color-scripts/
+    cp -r colorscripts $out/opt/shell-color-scripts/colorscripts
+    chmod +x colorscript.sh
+    cp colorscript.sh $out/bin/colorscript
+    substituteInPlace $out/bin/colorscript \
+      --replace "/opt/shell-color-scripts/colorscripts" "$out/opt/shell-color-scripts/colorscripts"
+    wrapProgram $out/bin/colorscript \
+      --prefix PATH : ${lib.makeBinPath [ runtimeShell ]}
+    runHook postInstall
+    '';
```
Comment (AndersonTorres) [inline pkgs/tools/misc/dt-shell-color-scripts/default.nix:36]: Back off indentation when closing.

```suggestion
  '';
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Review by AndersonTorres — CHANGES_REQUESTED** [review]
Code is OK.

`git rebase` all those commits into only one: `dt-shell-color-scripts: 2022-02-22`

**Comment by jakobneufeld** [comment]
Yeah i will do that thanks!

File: pkgs/tools/misc/dt-shell-color-scripts/default.nix [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    patchShebangs ./colorscripts
+    mkdir -p $out/bin
+    mkdir -p $out/opt/shell-color-scripts/
+    cp -r colorscripts $out/opt/shell-color-scripts/colorscripts
+    chmod +x colorscript.sh
+    cp colorscript.sh $out/bin/colorscript
+    substituteInPlace $out/bin/colorscript \
+      --replace "/opt/shell-color-scripts/colorscripts" "$out/opt/shell-color-scripts/colorscripts"
+    wrapProgram $out/bin/colorscript \
+      --prefix PATH : ${lib.makeBinPath [ runtimeShell ]}
```
Comment (AndersonTorres) [inline pkgs/tools/misc/dt-shell-color-scripts/default.nix:33]: Is there any reason to use it, after all? Reading the script, I have found no reason to call Bash or any runtime shell into its PATH.

**Comment by jakobneufeld** [comment]
@AndersonTorres I think I fixed everything you asked. Thanks so much for reviewing my package. I have never packages shell scripts in NixOS before, only rust programs which are easier.

**Comment by AndersonTorres** [comment]
Finally, fix the commit message:

`dt-shell-color-scripts: init at unstable-2022-02-22`

**Comment by jakobneufeld** [comment]
@AndersonTorres Finally ready! :)

**Pull request merged**